### ISSUE TITLE
Not installable on 11ty 3.0

### ISSUE THREAD

**Issue by MattWilcox** [photogabble/eleventy-plugin-interlinker#57]
```
npm i @photogabble/eleventy-plugin-interlinker

npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: @photogabble/eleventy-plugin-interlinker@1.0.6
npm error Found: @11ty/eleventy@3.0.0-alpha.4
npm error node_modules/@11ty/eleventy
npm error   dev @11ty/eleventy@"3.0.0-alpha.4" from the root project
npm error
npm error Could not resolve dependency:
npm error peerOptional @11ty/eleventy@">=2.0.0-beta.1" from @photogabble/eleventy-plugin-interlinker@1.0.6
npm error node_modules/@photogabble/eleventy-plugin-interlinker
npm error   @photogabble/eleventy-plugin-interlinker@"*" from the root project
npm error
npm error Conflicting peer dependency: @11ty/eleventy@2.0.1
npm error node_modules/@11ty/eleventy
npm error   peerOptional @11ty/eleventy@">=2.0.0-beta.1" from @photogabble/eleventy-plugin-interlinker@1.0.6
npm error   node_modules/@photogabble/eleventy-plugin-interlinker
npm error     @photogabble/eleventy-plugin-interlinker@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /Users/mattwilcox/.npm/_logs/2024-05-23T19_14_53_962Z-eresolve-report.txt
```

Just adding this here as I'd love to see a 3.0 compatible version (guess who's learning 11ty with 3.0 because he can just use ESM and not have to learn all the old-school node rubbish?)

**Comment by BravishkaSkytano** [comment]
As of now, I can install the plugin, but when I try to start the server I get these errors:

```
[11ty] Eleventy Error (CLI):
[11ty] 1. Error in your Eleventy config file 'eleventy.config.js'. (via EleventyConfigError)
[11ty] 2. `require("@11ty/eleventy")` is incompatible with Eleventy v3 and this version of Node. You have a few options:
[11ty]    1. (Easiest) Change the `require` to use a dynamic import inside of an asynchronous CommonJS configuration
[11ty]       callback, for example:
[11ty]
[11ty]       module.exports = async function {
[11ty]         const {EleventyRenderPlugin, EleventyI18nPlugin, EleventyHtmlBasePlugin} = await import("@11ty/eleventy");
[11ty]       }
[11ty]
[11ty]    2. (Easier) Update the JavaScript syntax in your configuration file from CommonJS to ESM (change `require`
[11ty]       to use `import` and rename the file to have an `.mjs` file extension).
[11ty]
[11ty]    3. (More work) Change your project to use ESM-first by adding `"type": "module"` to your package.json. Any
[11ty]       `.js` will need to be ported to use ESM syntax (or renamed to `.cjs`.)
[11ty]
[11ty]    4. (Short term workaround) Use the --experimental-require-module flag to enable this behavior. Read
[11ty]       more: https://nodejs.org/api/modules.html#loading-ecmascript-modules-using-require It is possible that the
[11ty]       newest version of Node has this enabled by default—you can try upgrading your version of Node.js.
```

I'm guessing the plugin hasn't been fully updated for Eleventy V3 yet.

**Comment by ZeFish** [comment]
Oh I'm getting that error also... don't know what to do at that point.

**Comment by carbontwelve** [comment]
Hey, yes this plugin has yet to be updated for V3. It's actually something I am working on this week as I have updated my website to V3 in order to test it with.

**Comment by carbontwelve** [comment]
@ZeFish @MattWilcox thank you for reporting. I will be working on this today, with a view of getting an RC release containing a fix ASAP. 

I believe it to be a case of using a tool like [duel](https://github.com/knightedcodemonkey/duel) in order to build the plugin as a dual package meaning it can be installed by people running CommonJS or ESM imports. Then I can update the 11ty version dependency to include version three.

**Comment by carbontwelve** [comment]
Hi all, I have just released version [v1.1.0-rc5](https://github.com/photogabble/eleventy-plugin-interlinker/releases/tag/v1.1.0-rc5) containing a bugfix for this issue and have that version running on my own website with Eleventy v3.0.0.
